sktime/registry/_lookup.py

```diff
@@ -134,20 +134,26 @@ def _is_estimator(name, klass):
     # packages
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", category=FutureWarning)
+        warnings.simplefilter("module", category=ImportWarning)
         for _, module_name, _ in pkgutil.walk_packages(path=[ROOT], prefix="sktime."):
 
             # Filter modules
             if _is_ignored_module(module_name) or _is_private_module(module_name):
                 continue
 
-            module = import_module(module_name)
-            classes = inspect.getmembers(module, inspect.isclass)
-
-            # Filter classes
-            estimators = [
-                (name, klass) for name, klass in classes if _is_estimator(name, klass)
-            ]
-            all_estimators.extend(estimators)
+            try:
+                module = import_module(module_name)
+                classes = inspect.getmembers(module, inspect.isclass)
+
+                # Filter classes
+                estimators = [
+                    (name, klass)
+                    for name, klass in classes
+                    if _is_estimator(name, klass)
+                ]
+                all_estimators.extend(estimators)
+            except ModuleNotFoundError as e:
```

Comment:
Could we be more specific on the exception catching? Since we're importing modules here, there may be any number of reasons why we see an `ModuleNotFoundError`. I would add an additional condition based on the error message.

Reply:
What kind of condition/reason do you imagine?  Does `ModuleNotFoundError` happen for reasons other than extra modules being nonpresent?  I always pause a little when acting on error messages because I don't know how non-english operating systems are handled by python ...

Reply:
yes, I also don't see what we would change/improve here, perhaps you can be more specific what you mean, @mloning?

Maybe @mloning means that the warning should be more specific to `all_estimators` rather than just a generic import error, so sth like "warning, estimators interfacing the following packages are not listed as they are not installed: [list of packages]"?

Reply:
Could also just add a comment making the purpose of the exception more clear.  Not sure where to put the trade-off between user friendliness and clarity for debugging.

Reply:
sure, I guess sth on what's trying to load the module and how to resolve it would help

Reply:
@xloem yes I also looked up what the best way is to handle this. I thought that it may be good to check for the specific error message that we expect from missing soft dependencies rather than the generic `ModuleNotFoundError`. I'm not sure about other possible reasons, perhaps if installations go wrong, missing compiled files, accidentally deleted files? Would the error message raised by sktime when a soft dependency is missing be language dependent? You can find it here: https://github.com/alan-turing-institute/sktime/blob/722a3705a14c9b164cb548a160db4d24b2fd037f/sktime/utils/validation/_dependencies.py#L25

Reply:
Incredibly relevant code.  Do you know if that function is called for every import here with a soft dependency?  Or would it be appropriate to call it manually in this function?

Reply:
We make sure that missing soft dependencies raise a more helpful error message and not just a generic module not found error. So I don't think we need to call it here, that should be taken care of in the respective modules. But I think we should check the expected error message when catching the exception.

Reply:
I'm testing this out a little towards implementing it.  I actually found there's a lot of use in catching errors _other_ than `ModuleNotFoundError`.  For example, on my system right now `ROCKETClassifier` is raising an `UnsupportedError` inside numba when imported, preventing enumeration of estimators.

Although `UnsupportedError` isn't specifically the case of a soft dependency missing, `all_estimators` is still most useful to a user if it can skip estimators that fail to import and return to the user a list of ones that presently work.

What are your thoughts on going the other way, catching `Exception` instead of `ModuleNotFoundError`, and displaying a cause-agnostic warning when an exception is caught?

Reply:
Thanks for adding the message based condition. 

Not sure about the other idea. For example, `numba` is a core dependency, so if that cannot be imported without errors I think I'd prefer raising the exception rather than a warning. It may be more confusing for users to see a warning there which makes it unclear whether this is something that needs fixing or not, compared to an error.

Reply:
hm, what's the logic that you would suggest, @mloning?

Reply:
I'm not quite setup to retest, but I think other parts of `numba` may have been working for me; and I think the other solution gives the user more flexibility.  Still, what you say sounds ok.